test/test_framework_test.go

```diff
@@ -2532,3 +2534,144 @@ func TestReplaceImports(t *testing.T) {
 
 	assert.Equal(t, expected, replacedCode)
 }
+
+func TestCoverageReport(t *testing.T) {
+	t.Parallel()
+
+	const fooContract = `
+        pub contract FooContract {
+            pub let specialNumbers: {Int: String}
+
+            init() {
+                self.specialNumbers = {
+                    1729: "Harshad",
+                    8128: "Harmonic",
+                    41041: "Carmichael"
+                }
+            }
+
+            pub fun addSpecialNumber(_ n: Int, _ trait: String) {
+                self.specialNumbers[n] = trait
+            }
+
+            pub fun getIntegerTrait(_ n: Int): String {
+                if n < 0 {
+                    return "Negative"
+                } else if n == 0 {
+                    return "Zero"
+                } else if n < 10 {
+                    return "Small"
+                } else if n < 100 {
+                    return "Big"
+                } else if n < 1000 {
+                    return "Huge"
+                }
+
+                if self.specialNumbers.containsKey(n) {
+                    return self.specialNumbers[n]!
+                }
+
+                return "Enormous"
+            }
+        }
+    `
+
+	code := `
+        import FooContract from "./FooContract.cdc"
+
+        pub let foo = FooContract()
+
+        pub fun testGetIntegerTrait() {
+            // Arrange
+            let testInputs: {Int: String} = {
+                -1: "Negative",
+                0: "Zero",
+                9: "Small",
+                99: "Big",
+                999: "Huge",
+                1001: "Enormous",
+                1729: "Harshad",
+                8128: "Harmonic",
+                41041: "Carmichael"
+            }
+
+            for input in testInputs.keys {
+                // Act
+                let result = foo.getIntegerTrait(input)
+
+                // Assert
+                assert(result == testInputs[input])
+            }
+        }
+
+        pub fun testAddSpecialNumber() {
+            // Act
+            foo.addSpecialNumber(78557, "Sierpinski")
+
+            // Assert
+            assert("Sierpinski" == foo.getIntegerTrait(78557))
+        }
+    `
+
+	importResolver := func(location common.Location) (string, error) {
+		return fooContract, nil
```

Comment:
Maybe only return this for `StringLocation("./FooContract.cdc")`

Reply:
Good catch :+1:
Fixed in https://github.com/onflow/cadence-tools/pull/78/commits/a7caee505d22fcc903ce4c0e752287e5df5244ce